packages/web/src/pages/sign-up-page/components/layout.tsx

```diff
@@ -11,73 +13,109 @@ import {
   continue: 'Continue'
 }
 
 type PageProps = FlexProps & {
   as?: ComponentType<any>
+  centered?: boolean
 }
 
 export const Page = (props: PageProps) => {
+  const { centered, children, as, ...other } = props
   const { isMobile } = useMedia()
+
+  const childrenArray = Children.toArray(children)
+  const footer = childrenArray.pop()
+
+  const layoutProps: FlexProps = {
+    direction: 'column',
+    h: '100%',
+    gap: '2xl',
+    ph: isMobile ? 'l' : '2xl',
+    pv: 'xl'
+  }
+
+  if (centered) {
+    return (
+      <Flex h='100%' direction='column' alignItems='center' as={as}>
+        <Flex
+          {...layoutProps}
+          {...other}
+          alignSelf='center'
+          css={!isMobile && { maxWidth: 610 }}
+        >
+          {childrenArray}
+        </Flex>
+        {footer}
+      </Flex>
+    )
+  }
+
   return (
-    <Flex
-      direction='column'
-      h='100%'
-      gap='2xl'
-      ph={isMobile ? 'l' : '2xl'}
-      pv='2xl'
-      {...props}
-    />
+    <Flex as={as} {...layoutProps} {...other}>
+      {children}
+    </Flex>
   )
 }
 
 type HeadingProps = {
   prefix?: ReactNode
-  heading: string
-  description: string
-}
+  postfix?: ReactNode
+  heading: ReactNode
+  description?: ReactNode
+  centered?: boolean
+} & Omit<FlexProps & BoxProps, 'prefix'>
 
-export const Heading = (props: HeadingProps) => {
-  const { prefix, heading, description } = props
-  const { isMobile } = useMedia()
-  return (
-    <Flex gap={isMobile ? 's' : 'l'} direction='column'>
-      {prefix}
-      <Text
-        color='heading'
-        size={isMobile ? 'm' : 'l'}
-        strength='default'
-        variant='heading'
+export const Heading = forwardRef<HTMLDivElement, HeadingProps>(
+  (props, ref) => {
+    const { prefix, heading, description, postfix, centered, ...other } = props
+    const { isMobile } = useMedia()
+    return (
+      <Flex
+        ref={ref}
+        gap={isMobile ? 's' : 'l'}
+        direction='column'
+        alignItems={centered ? 'center' : undefined}
+        {...other}
       >
-        {heading}
-      </Text>
-      <Text size={isMobile ? 'm' : 'l'} variant='body'>
-        {description}
-      </Text>
-    </Flex>
-  )
-}
+        {prefix}
+        <Text variant='heading' color='accent' size={isMobile ? 'm' : 'l'}>
+          {heading}
+        </Text>
+        {description ? (
+          <Text size={isMobile ? 'm' : 'l'} variant='body'>
+            {description}
+          </Text>
+        ) : null}
+        {postfix}
+      </Flex>
+    )
+  }
+)
 
 type PageFooterProps = {
   prefix?: ReactNode
   postfix?: ReactNode
   buttonProps?: ButtonProps
-}
+  centered?: boolean
+} & Omit<ContinueFooterProps, 'prefix'>
 
 export const PageFooter = (props: PageFooterProps) => {
-  const { prefix, postfix, buttonProps } = props
+  const { prefix, postfix, buttonProps, centered, ...other } = props
+  const { isMobile } = useMedia()
 
   return (
-    <ContinueFooter>
+    <ContinueFooter {...other}>
```

Comment:
should these just be one component? Do we use `ContinueFooter` outside of `PageFooter`?

Reply:
great call, meant to clean this up after they all got updated